nncf/sparsity/schedulers.py

```diff
@@ -236,3 +239,28 @@ def current_sparsity_level(self):
     @property
     def target_sparsity_level(self):
         return self.sparsity_levels[-1]
+
+
+@SPARSITY_SCHEDULERS.register("per_layer")
+class PerLayerSparsityScheduler(SparsityScheduler):
+    def __init__(self, sparsity_algo, params):
+        super().__init__(sparsity_algo, params)
+        self.sparsified_modules_info_by_modules_name = {}
+        for minfo in self.algo.sparsified_module_info:
+            self.sparsified_modules_info_by_modules_name[minfo.module_name] = minfo
+        self.layered_sparsity_control_algo = None
+        self.initialize()
+
+    def initialize(self):
+        for minfo in self.algo.sparsified_module_info:
+            self._set_sparsity_level_for_module(minfo.module_name, self.initial_sparsity)
+
+    def _set_sparsity_level_for_module(self, module_name: str, sparsity_level: float):
+        minfo = [self.sparsified_modules_info_by_modules_name[module_name]]
+        self.algo.set_sparsity_level_for_module(sparsity_level, minfo)
+
+    def epoch_step(self, epoch=None):
+        self.algo.do_specific_setting_after_each_epoch()
```

Comment:
Why do you call the batchnorn adaptation algorithm after each epoch?

Reply:
This was incorrect. Fixed.